Intermittent follow_replication test failure

Please [read these guidelines](http://ibm.biz/cdt-issue-guide) before opening an issue.

<!-- Issues will be CLOSED IMMEDIATELY if the following template is not completed. -->

## Bug Description

### 1. Steps to reproduce and the simplest code sample possible to demonstrate the issue

Run `tests.unit.replicator_tests.ReplicatorTests.test_follow_replication`

### 2. What you expected to happen

Test to pass.

### 3. What actually happened

Test failed with:
```
[2020-04-14T12:10:17.930Z] FAIL: Test that follow_replication(...) properly iterates updated
[2020-04-14T12:10:17.930Z]----------------------------------------------------------------------
[2020-04-14T12:10:17.930Z] Traceback (most recent call last):
[2020-04-14T12:10:17.930Z]   File "/home/jenkins/workspace/_cloudant_prepare-2.13.0-release/tests/unit/replicator_tests.py", line 417, in test_follow_replication
[2020-04-14T12:10:17.930Z]     self.assertIn(doc.get(state_key), valid_states)
[2020-04-14T12:10:17.930Z] AssertionError: 'initializing' not found in ('completed', 'error', 'triggered', 'running', None)
```

## Environment details
- Version(s) that are affected by this issue.
    > master
- Python version
    > 3.5.2


Looks like we need to add `initializing` to the list of valid states.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,6 +1,7 @@
 # UNRELEASED
 - [REMOVED] Removed Python 2 compatibility from the supported environments.
 - [FIXED] Fixed the documentation for `bookmarks`.
+- [FIXED] Also exit `follow_replication` for `failed` state.
 
 # 2.14.0 (2020-08-17)
 

--- a/src/cloudant/replicator.py
+++ b/src/cloudant/replicator.py
@@ -193,7 +193,12 @@ class Replicator(object):
             repl_doc, state = update_state()
             if repl_doc:
                 yield repl_doc
-            if state is not None and state in ['error', 'completed']:
+            # This is a little awkward, since 2.1 the terminal states are
+            # "failed" and "completed", so those should be the exit states, but
+            # for backwards compatibility with older versions "error" is also
+            # needed. The code has always exited for "error" state even long
+            # after 2.1 was available so that behaviour is retained.
+            if state is not None and state in ['error', 'failed', 'completed']:
                 return
 
             # Now listen on changes feed for the state
@@ -202,7 +207,8 @@ class Replicator(object):
                     repl_doc, state = update_state()
                     if repl_doc is not None:
                         yield repl_doc
-                    if state is not None and state in ['error', 'completed']:
+                    # See note about these states
+                    if state is not None and state in ['error', 'failed', 'completed']:
                         return
 
     def stop_replication(self, repl_id):

--- a/tests/unit/replicator_tests.py
+++ b/tests/unit/replicator_tests.py
@@ -322,12 +322,14 @@ class ReplicatorTests(UnitTestDbBase):
         )
         self.replication_ids.append(repl_id)
         repl_state = None
-        valid_states = ['completed', 'error', 'triggered', 'running', None]
+        # note triggered is for versions prior to 2.1
+        valid_states = ['completed', 'error', 'initializing', 'triggered', 'pending', 'running', 'failed', 'crashing', None]
         finished = False
+        # Wait for 5 minutes or a terminal replication state
         for _ in range(300):
             repl_state = self.replicator.replication_state(repl_id)
             self.assertTrue(repl_state in valid_states)
-            if repl_state in ('error', 'completed'):
+            if repl_state in ('error', 'failed', 'completed'):
                 finished = True
                 break
             time.sleep(1)
@@ -407,7 +409,8 @@ class ReplicatorTests(UnitTestDbBase):
             repl_id
         )
         self.replication_ids.append(repl_id)
-        valid_states = ('completed', 'error', 'triggered', 'running', None)
+        # note triggered is for versions prior to 2.1
+        valid_states = ['completed', 'error', 'initializing', 'triggered', 'pending', 'running', 'failed', 'crashing', None]
         repl_states = []
         if 'scheduler' in self.client.features():
             state_key = 'state'

